database/migrations/0008_purchases_paypal_apple_google.sql

```diff
@@ -0,0 +1,46 @@
+-- 0008
+
+CREATE TABLE account_paypal_order (
+    account_id INTEGER NOT NULL REFERENCES account(id) ON DELETE CASCADE,
+    payment_id VARCHAR PRIMARY KEY,
+    state VARCHAR
+);
+
+CREATE TABLE account_app_store_purchase (
+    account_id INTEGER NOT NULL REFERENCES account(id) ON DELETE CASCADE,
+    transaction_id VARCHAR PRIMARY KEY,
+    cancellation_date VARCHAR,
+    cancellation_date_ms VARCHAR,
+    cancellation_date_pst VARCHAR,
+    cancellation_reason VARCHAR,
+    expires_date VARCHAR,
+    expires_date_ms VARCHAR,
```

Comment:
What would the format be for expires_date, expires_date_ms, etc? Can you not use any of the time data types?

Reply:
That's a good catch, and if I recall I didn't specify column types except `varchar` because the `account_app_store_purchase` table is simply saving all the data of the purchase receipt that our api receives from Apple. I don't actually have a use case for all of this data, but it seemed good to save it just in case, but I've never needed to query any of these payment receipt tables in 4 years.

Sooo...I was just stuffing the App Store receipt data in as varchars, thinking we don't have a programmatic purpose for it, and I didn't want to deal with type issues from the App Store. I'm opposed to making these columns properly typed, but I also don't have a use case in mind if they are typed properly.

Reply:
Ha! Sorry, I laughed reading this. I think this is sound then. Save the energy and focus it elsewhere. Ignore my comment !